### ISSUE TITLE
Extensions can only declare one theme

### ISSUE THREAD

**Issue by steff456** [jupyterlab/jupyterlab#14202]
### Problem

My main goal is to be able to maintain multiple accessible and WCAG compliant JupyterLab themes from one package, but apparently it is not possible. I've tried multiple configurations from a single extension without any luck because the API for themes is too restrictive, specially the `themePath` option.

### Proposed Solution

Adding support to multiple themes from one single extension should be possible. This should be fixable if `themePath` option gets extended to a list or object that supports multiple directories to load multiple `index.css`. 

From my investigation we should need to do the changes in the following places,

https://github.com/jupyterlab/jupyterlab/blob/a396badbb6a3c840bdc088b22463fbcb42376bbe/builder/src/build.ts#L85

https://github.com/jupyterlab/jupyterlab/blob/a396badbb6a3c840bdc088b22463fbcb42376bbe/jupyterlab/commands.py#L2077

https://github.com/jupyterlab/jupyterlab/blob/a396badbb6a3c840bdc088b22463fbcb42376bbe/builder/metadata_schema.json#L76

### Additional context

I would be able to make the described changes and send a PR for this feature, please let me know if there's anything else I should be aware of and if there was a preference or history I'm unaware of.

**Comment by krassowski** [comment]
For some more context existing extensions solve this by bundling multiple packages as for example https://github.com/gt-coar/jupyterlab-gt-coar-theme.

If themePath were a list how would JupyterLab associate the theme with is name and possibly other metadata?

**Comment by steff456** [comment]
If we want to store more meta data for each theme I would think to do an object and we can define an interface that encapsules all the info we want.

I'm seeing the example that you have, and it works. At the end we can also add some official documentation on how to do it because with the current docs they assume that the developer only wants to create one theme per extension. 

I'm not sure if we want to pursue the option of modifying the extension API to register multiple themes or if I just go the workspace way and document it in the docs. Let me know what you think and I can do either option.

**Comment by krassowski** [comment]
>  Let me know what you think and I can do either option.

This might be worth bringing up on Wednesday call. I do not have a strong preference, but documenting this along with pointing to a maintained example (such as the accessible themes extension) would be rather uncontroversial. Deciding on adding another source of truth about extension metadata may need a bit more discussion.

Tagging @bollwyvl who may have an opinion.
